utils/crypto/bls/signer/rpcsigner/client.go

```diff
@@ -16,50 +21,73 @@ import (
 var _ bls.Signer = (*Client)(nil)
 
 type Client struct {
-	client pb.SignerClient
-	pk     *bls.PublicKey
+	client     pb.SignerClient
+	pk         *bls.PublicKey
+	connection *grpc.ClientConn
 }
 
-func NewClient(ctx context.Context, conn *grpc.ClientConn) (*Client, error) {
+func NewClient(ctx context.Context, url string) (*Client, error) {
+	// TODO: figure out the best parameters here given the target block-time
+	opts := grpc.WithConnectParams(grpc.ConnectParams{
+		Backoff: backoff.DefaultConfig,
+		// same as grpc default
+		MinConnectTimeout: 20 * time.Second,
+	})
+
+	// the rpc-signer client should call a proxy server (on the same machine) that forwards
+	// the request to the actual signer instead of relying on tls-credentials
+	conn, err := grpc.NewClient(url, opts, grpc.WithTransportCredentials(insecure.NewCredentials()))
+	if err != nil {
+		return nil, fmt.Errorf("failed to create rpc signer client: %w", err)
+	}
+
 	client := pb.NewSignerClient(conn)
 
 	pubkeyResponse, err := client.PublicKey(ctx, &pb.PublicKeyRequest{})
 	if err != nil {
-		return nil, err
+		return nil, errors.Join(fmt.Errorf("failed to get pubkey response: %w", err), conn.Close())
 	}
 
 	pkBytes := pubkeyResponse.GetPublicKey()
 	pk, err := bls.PublicKeyFromCompressedBytes(pkBytes)
 	if err != nil {
-		return nil, err
+		return nil, errors.Join(fmt.Errorf("failed to uncompress public key bytes: %w", err), conn.Close())
 	}
 
 	return &Client{
-		client: client,
-		pk:     pk,
+		client:     client,
+		pk:         pk,
+		connection: conn,
 	}, nil
 }
 
 func (c *Client) PublicKey() *bls.PublicKey {
 	return c.pk
 }
 
+// Sign a message. The [Client] already handles transient connection errors.
```

Comment:
I would move the comment w.r.t transient conn errors to `NewClient` or `Client` since it's applicable to the type and not just `Sign`